src/views/flamechart-view-container.tsx

```diff
@@ -28,51 +26,34 @@ interface FlamechartSetters {
   setSelectedNode: (node: CallTreeNode | null) => void
 }
 
-interface WithFlamechartContext<T> {
-  profileIndex: number
-  args: {
-    id: FlamechartID
-  } & T
-}
+const {
+  setHoveredNode,
+  setLogicalSpaceViewportSize,
+  setConfigSpaceViewportRect,
+  setSelectedNode,
+} = actions.flamechart
 
 export function useFlamechartSetters(id: FlamechartID, profileIndex: number): FlamechartSetters {
-  function useActionCreatorWithIndex<T, U>(
-    actionCreator: ActionCreator<WithFlamechartContext<U>>,
-    map: (t: T) => U,
-  ): (t: T) => void {
-    const callback = useCallback(
-      (t: T) => {
-        const args = Object.assign({}, map(t), {id})
-        return actionCreator({profileIndex, args})
-      },
-      [actionCreator, map],
```

Comment:
The actual bug was that `profileIndex` and `id` were missing from this array. I'm not sure why this wasn't caught by the exhaustive deps check.